localgov_news.module

```diff
@@ -24,6 +24,14 @@ function localgov_news_theme($existing, $type, $theme, $path) {
   ];
 }
 
+/**
+ * Implements hook_page_attachments().
+ */
+function localgov_news_page_attachments(array &$attachments) {
+  // Attach news styling library.
+  $attachments['#attached']['library'][] = 'localgov_news/news-styling';
```

Comment:
Does this mean that the library gets attached to every page once the news module is installed? 

If so, we should amend it to only attach the new styling to pages where the content type is news-article and/or newsroom.

Reply:
> Does this mean that the library gets attached to every page once the news module is installed?
> 
> If so, we should amend it to only attach the new styling to pages where the content type is news-article and/or newsroom.

My intention was to remove all styling from the news module. There's only a very basic layout for the newsroom, and the category styling in the module, so it makes little sense to have it here?

This is definitely an area where some standards for the project would help.

Reply:
@danchamp if you do need to keep some styles with this module you can use `hook_ENTITY_TYPE_view_alter` with a condition on the bundle only attach to news nodes.

Reply:
Thanks @j4-m. Is there any convention for when to include CSS (or what CSS to include) within a module?

I can see that most localgov_ modules have none or very little, except step-by-step and subsites both of which have interface elements.

Reply:
@danchamp generally, and particularly with contrib., a module would only provide CSS if it implements it's [theme hook](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Render%21theme.api.php/function/hook_theme/8.2.x) and this is the case us [step-by-step](https://github.com/localgovdrupal/localgov_step_by_step/blob/master/localgov_step_by_step.module#L21) and [subsites](https://github.com/localgovdrupal/localgov_subsites/blob/master/localgov_subsites.module#L14).

If your module is not implementing hook theme, i.e. not providing a new template, then ideally you would be making use of templates provided by core or contrib., via field formatters etc., and leaving the styles to the theme.

LocalGov is a bit of a different beast however and really sure what the precedent is for this sort of thing. @markconroy, is this the sort of thing you're hoping to work through in the meeting next Thursday?

Hope this helps @danchamp  Do let me know if it's not clear.